Routing/Loader/RestYamlCollectionLoader.php

```diff
@@ -60,11 +61,25 @@ public function load($file, $type = null)
     {
         $path = $this->locator->locate($file);
 
-        $config = Yaml::parse(file_get_contents($path));
+        try {
+            $config = Yaml::parse(file_get_contents($path));
+        } catch (ParseException $e) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not contain valid YAML.', $path), 0, $e);
+        }
 
         $collection = new RouteCollection();
         $collection->addResource(new FileResource($path));
 
+        // empty file
+        if (null === $config) {
+            return $collection;
+        }
+
+        // not an array
+        if (!is_array($config)) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" must contain a YAML array.', $path));
```

Comment:
it is called `mapping` in Yaml. maybe we can replace that by `a mapping (an array)`?